dotnet/src/SemanticKernel.Abstractions/Orchestration/IKernelContext.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using Microsoft.SemanticKernel.SkillDefinition;
+
+namespace Microsoft.SemanticKernel.Orchestration;
+
+/// <summary>
+/// Kernel context reference
+/// </summary>
+public interface IKernelContext
```

Comment:
This name is pretty generic. I'd recommend using `IKernelExecutionContext` to make it clear the intended use. 

I'd also recommend adding a better summary which explains the lifecycle.

Reply:
Agree, changed the both implementation and interface names to `IKernelExecutionContext`